### PR TITLE
Init 2.1 build of InfluxDB

### LINKED ISSUE #534: Init 2.1 `influxdb` Dockerfile
1. Copy the `2.0` tree to a new `2.1` directory
2. Modify the `Dockerfile`s under `2.1` to pull the new CLI from `influx-cli`
3. Adjust our `influxdb` e2e tests to cover both `2.0` and `2.1`

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #534

For now the `2.1` builds pull down `2.0.9` of the server and `2.1.1` of the CLI. Once v2.1 of the server is released, this PR should be updated.

### CODE REVIEW

File: influxdb/2.1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	gpg --batch --keyserver hkps://keys.openpgp.org --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 && \
+	gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu && \
+	gpgconf --kill all && \
+	rm -rf "$GNUPGHOME" /usr/local/bin/gosu.asc && \
+	chmod +x /usr/local/bin/gosu && \
+	gosu --version && \
+	gosu nobody true
+
+# Install the infuxd server
+ENV INFLUXDB_VERSION 2.0.9
```
Comment (codyshepherd) [inline influxdb/2.1/Dockerfile:25]: You'll want to update this

File: influxdb/2.1/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    gpg --batch --verify influxdb2-client-${INFLUX_CLI_VERSION}-linux-${ARCH}.tar.gz.asc influxdb2-client-${INFLUX_CLI_VERSION}-linux-${ARCH}.tar.gz && \
+    tar xzf influxdb2-client-${INFLUX_CLI_VERSION}-linux-${ARCH}.tar.gz && \
+    cp influxdb2-client-${INFLUX_CLI_VERSION}-linux-${ARCH}/influx /usr/local/bin/influx && \
+    rm -rf "$GNUPGHOME" influxdb2.key influxdb2-client-${INFLUX_CLI_VERSION}-linux-${ARCH}* && \
+    influx version
+
+# Create standard directories expected by the entry-point.
+RUN mkdir /docker-entrypoint-initdb.d && \
+    mkdir -p /var/lib/influxdb2 && \
+    chown -R influxdb:influxdb /var/lib/influxdb2 && \
```
Comment (codyshepherd) [inline influxdb/2.1/Dockerfile:67]: Do similar chown steps need to be done for the cli and `influx` as well in this case where the cli is a separate install?

File: influxdb/2.1/alpine/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	gpg --batch --keyserver hkps://keys.openpgp.org --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4; \
+	gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu; \
+	gpgconf --kill all; \
+	rm -rf "$GNUPGHOME" /usr/local/bin/gosu.asc; \
+	chmod +x /usr/local/bin/gosu; \
+	gosu --version; \
+	gosu nobody true
+
+# Install the infuxd server
+ENV INFLUXDB_VERSION 2.0.9
```
Comment (codyshepherd) [inline influxdb/2.1/alpine/Dockerfile:34]: This will also need to get updated to 2.1

File: influxdb/2.1/Dockerfile (hunk 1 above)
Comment (codyshepherd) [reply influxdb/2.1/Dockerfile:25]: Oh I see the PR summary now. Ok, we can leave this until the 2.1 release.

File: influxdb/2.1/Dockerfile (hunk 2 above)
Comment (danxmoran) [reply influxdb/2.1/Dockerfile:67]: I think this still covers both. The CLI's only file-on-disk is `/etc/influxdb2/influx-configs`

**Comment by wbaker85** [comment]
Updated the docker files to point at the `2.1.0` release.

**Review by wbaker85 — APPROVED** [review]

**Pull request merged**